app_server/app/controllers/AdminController.js

```diff
@@ -288,12 +305,21 @@ var encrypt = function(password) {
 };
 
 var updateCache = function(request, account, callback) {
-  request.server.app.cache.set(account.userId, account, 0, function (err) {
-    if (err) {
-      logger.error(err);
+  request.server.app.cache.get(account.userId, function(err, cached) {
+    if (cached && cached.item) {
+      // merge with cached version, with new taking priority
+      Object.keys(cached.item)
+            .filter((k) => !account.hasOwnProperty(k))
```

Comment:
? It's filtering out matching properties from `account`, though?
Like...
```
cached.item {'a': 1, 'b': 1, 'c': 1}
account {'a': 1, 'b': 1}
filtered {'c': 1}
```
This segment is merging the `cached.item` and `account` objects.

Did I misunderstand something?